centrifuge-app/src/components/InvestRedeem/InvestRedeemDrawer.tsx

```diff
@@ -72,14 +58,78 @@ const TokenPriceChart = React.memo(function TokenPriceChart({
 
   const { poolStates: dailyPoolStates } = useDailyPoolStates(poolId, new Date(dateFrom)) || {}
 
+  if (!dailyPoolStates?.length) return
+
+  return (
+    <Drawer isOpen={open} onClose={onClose}>
+      <LoadBoundary>
+        <InvestRedeem poolId={poolId} trancheId={trancheId} defaultView={defaultView} />
+      </LoadBoundary>
+      <LoadBoundary>
+        <Stack gap={12} borderColor="rgba(0,0,0,0.08)" borderWidth="1px" borderStyle="solid" borderRadius="8px" p={2}>
+          <Box display="flex" justifyContent="space-between" alignItems="center" mb={2}>
+            <Text variant="heading6" color="textPrimary" fontWeight={600}>
+              Performance
+            </Text>
+            <Tabs selectedIndex={index} onChange={(index) => setIndex(index)}>
+              <TabsItem styleOverrides={{ padding: '4px' }} showBorder>
+                Price
+              </TabsItem>
+              <TabsItem styleOverrides={{ padding: '4px' }} showBorder>
+                APY
+              </TabsItem>
+            </Tabs>
+          </Box>
+          <TokenPriceChart
+            poolId={poolId}
+            trancheId={trancheId}
+            dailyPoolStates={dailyPoolStates}
+            filter={filter}
+            setFilter={setFilter}
+            index={index}
+          />
+        </Stack>
+      </LoadBoundary>
+    </Drawer>
+  )
+}
+
+const TokenPriceChart = React.memo(function TokenPriceChart({
+  poolId,
+  trancheId,
+  dailyPoolStates,
+  filter,
+  setFilter,
+  index,
+}: {
+  poolId: string
+  trancheId: string
+  dailyPoolStates: DailyPoolStateProps[]
+  filter: FilterOptions
+  setFilter: any
+  index: number
+}) {
+  const pool = usePool(poolId)
+
+  const apy = {
+    '30days': 'yield30DaysAnnualized',
+    '90days': 'yield90DaysAnnualized',
+    YTD: 'yieldYTD',
+  }
+
   const data = React.useMemo(() => {
     const tokenData =
-      dailyPoolStates?.map((state) => {
-        return { price: state.tranches[trancheId].price?.toFloat() || 0, day: new Date(state.timestamp) }
+      dailyPoolStates?.map((state: DailyPoolStateProps) => {
```

Comment:
```suggestion
      dailyPoolStates?.map((state) => {
```
again this type should be interfered if `dailyPoolStates` is type properly